handlers/shuttleadapter.go

```diff
@@ -0,0 +1,39 @@
+package handlers
+
+import (
+	"context"
+
+	servicebus "github.com/Azure/azure-service-bus-go"
+	"github.com/Azure/go-shuttle/message"
+	"github.com/devigned/tab"
+)
+
+var _ servicebus.Handler = (*ShuttleAdapter)(nil)
+
+// ShuttleAdapter wraps a go-shuttle message.Handler in a servicebus.Handler.
+// It allows the consumer to write go-shuttle handler func
+// that uses the go-shuttle semantic and message disposition API
+type ShuttleAdapter struct {
```

Comment:
WDYT of the adapter or some other structure having a slice of middleware? Then when you actually want to materialize the final function you can build the handler from the stack of middleware like: https://github.com/devigned/pub/blob/8124492a2e15b32f2d0a55d9b6e7a71f33ed4478/pkg/partner/client.go#L557-L600.

The upside to this design is that you don't have to decorate next all over the place.

Reply:
Took the discussion offline.
Basically there is 2 approaches to middleware pipeline: 
- having an explicit pipeline that manage the flow from a list of middleware
- or the one I did, which chains the middlewares explicitly, where they call the next one each time.

I chose to chain them, because I don't think the middleware complexity is going to grow more than what it is now.
If it does, we can reconsider, but I think this works well, so I'd like to keep it this way.